pkg/supervisor/supervisor.go

```diff
@@ -130,14 +142,33 @@ func (s *supervisor) Start(ctx context.Context) {
 	s.reflectorCtl.Start(ctx)
 	defer events.Log("Stopped Supervisor")
 	for {
+		err := func() error {
+			// If the ledger latency is too much, temporarily stop uploading snapshots.
+			// We need to first catch up, or else we'll upload snapshots that are out-of-date
+			// which would put a significant amount of load on the exective because every new
+			// reflector will have to sync a potentially very large chunk of the DML ledger.
+			latency, err := s.reader.GetLedgerLatency(ctx)
+			if err != nil {
+				return err
+			}
+			isAcceptableLatency := s.minimumLedgerLatency > latency
+			stats.Set("is_acceptable_latency", isAcceptableLatency)
+			stats.Set("latency", latency)
+
+			if !isAcceptableLatency {
+				return nil
```

Comment:
Would recommend logging here that we are skipping snapshotting the ldb, and include the latency (and threshold we compared it to) in the log message.